tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -88,22 +89,23 @@ internal static string GetOperationName(Tracer tracer, string spanKind)
             };
         }
 
-        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IDictionary<string, object> headers)
+        internal static void SetDataStreamsCheckpointOnProduce(Tracer tracer, Span span, RabbitMQTags tags, IBody body, IBasicProperties basicProperties)
         {
             var dataStreamsManager = tracer.TracerManager.DataStreamsManager;
-            if (dataStreamsManager == null || !dataStreamsManager.IsEnabled || headers == null)
+            if (dataStreamsManager == null || basicProperties == null || !dataStreamsManager.IsEnabled || basicProperties.Headers == null)
             {
                 return;
             }
 
             try
             {
-                var headersAdapter = new RabbitMQHeadersCollectionAdapter(headers);
+                var headersAdapter = new RabbitMQHeadersCollectionAdapter(basicProperties.Headers);
                 var edgeTags = string.IsNullOrEmpty(tags.Exchange) ?
                                    // exchange can be empty for "direct"
                                    new[] { "direction:out", $"topic:{tags.Queue ?? tags.RoutingKey}", "type:rabbitmq" } :
                                    new[] { "direction:out", $"exchange:{tags.Exchange}", $"has_routing_key:{!string.IsNullOrEmpty(tags.RoutingKey)}", "type:rabbitmq" };
-                span.SetDataStreamsCheckpoint(dataStreamsManager, CheckpointKind.Produce, edgeTags);
+                // size of headers is ignored for now
```

Comment:
Is that because it's hard to calculate?

Reply:
I'll add the logic similar to Kafka payload size (i.e. size of objects in bytes). Not sure what would be the best way of reusing the code between integrations....

Reply:
I see - mostly wanted to make sure that you didn't mean to update this now and had forgotten it 🙂 If it's not easy to reuse between integrations, I wouldn't try at this stage, copy paste what you need and we can abstract it later if it makes sense!

Reply:
I guess this comment is out of date now? 🙂 
```suggestion
```